packages/@aws-cdk/custom-resources/lib/aws-custom-resource/runtime/index.ts

```diff
@@ -90,14 +90,14 @@ export async function handler(event: AWSLambda.CloudFormationCustomResourceEvent
     let physicalResourceId: string;
     switch (event.RequestType) {
       case 'Create':
-        physicalResourceId = event.ResourceProperties.Create?.physicalResourceId ??
-                             event.ResourceProperties.Update?.physicalResourceId ??
-                             event.ResourceProperties.Delete?.physicalResourceId ??
+        physicalResourceId = event.ResourceProperties.Create?.physicalResourceId.id ??
+                             event.ResourceProperties.Update?.physicalResourceId.id ??
+                             event.ResourceProperties.Delete?.physicalResourceId.id ??
```

Comment:
Should be `physicalResourceId?.literalId`

Reply:
Done